SmartDeviceLink_Example/ButtonManager.swift

```diff
@@ -0,0 +1,159 @@
+        let imageOffState = SDLSoftButtonState(stateName: ToggleSoftButtonImageOffState, text: nil, image: UIImage(named: LaptopIconImageName))
+        let textOnState = SDLSoftButtonState(stateName: ToggleSoftButtonTextOnState, text: ToggleSoftButtonTextTextOnText, image: nil)
+        let textOffState = SDLSoftButtonState(stateName: ToggleSoftButtonTextOffState, text: ToggleSoftButtonTextTextOffText, image: nil)
+        let softButton: SDLSoftButtonObject? = SDLSoftButtonObject(name: ToggleSoftButton, states: [imageOnState, imageOffState, textOnState, textOffState], initialStateName: imageOnState.name) { [unowned self] (buttonPress, buttonEvent) in
+            guard buttonPress != nil else { return }
+            self.toggleEnabled = !self.toggleEnabled
+        }
+
+        guard softButton != nil else { assert(false, "Button was not created successfully") }
+        return softButton!
+    }
+
+    /// Returns a soft button that toggles the textfield visibility state for the SDL UI. The button's text toggles based on the current text visibility.
+    ///
+    /// - Returns: A soft button
+    func softButtonTextVisible() -> SDLSoftButtonObject {
+        let textVisibleState = SDLSoftButtonState(stateName: TextVisibleSoftButtonTextOnState, text: TextVisibleSoftButtonTextOnText, artwork: nil)
+        let textNotVisibleState = SDLSoftButtonState(stateName: TextVisibleSoftButtonTextOffState, text: TextVisibleSoftButtonTextOffText, image: nil)
+        let softButton: SDLSoftButtonObject? = SDLSoftButtonObject(name: TextVisibleSoftButton, states: [textVisibleState, textNotVisibleState], initialStateName: textVisibleState.name) { [unowned self] (buttonPress, buttonEvent) in
+            guard buttonPress != nil else { return }
+            self.textEnabled = !self.textEnabled
+
+            // Update the button state
+            let softButton = self.sdlManager.screenManager.softButtonObjectNamed(TextVisibleSoftButton)
+            softButton?.transitionToNextState()
+        }
+
+        guard softButton != nil else { assert(false, "Button was not created successfully") }
+        return softButton!
+    }
+
+    /// Returns a soft button that toggles the image visibility state for the SDL UI. The button's text toggles based on the current image visibility.
+    ///
+    /// - Returns: A soft button
+    func softButtonImagesVisible() -> SDLSoftButtonObject {
+        let imagesVisibleState = SDLSoftButtonState(stateName: ImagesVisibleSoftButtonImageOnState, text: ImagesVisibleSoftButtonImageOnText, image: nil)
+        let imagesNotVisibleState = SDLSoftButtonState(stateName: ImagesVisibleSoftButtonImageOffState, text: ImagesVisibleSoftButtonImageOffText, image: nil)
+        let softButton: SDLSoftButtonObject? = SDLSoftButtonObject(name: ImagesVisibleSoftButton, states: [imagesVisibleState, imagesNotVisibleState], initialStateName: imagesVisibleState.name) { [unowned self] (buttonPress, buttonEvent) in
```

Comment:
This init object doesn't create a nullable object. The below guard is unnecessary.